From 7aebfaf6b55bf6fdbacde684219a073e2854eb09 Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Tue, 2 Apr 2013 15:27:01 +0800
Subject: [PATCH 181/231] ENGR00256933 ASoC: WM8962: Add delay after
 FLL-enable

  There's a start-up time after FLL_ENA bit is set. While the driver didn't
run the delay code which already exists in the set_fll() function.
  This patch let the driver run into the delay section, and made it work more
perfectly.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
---
 sound/soc/codecs/wm8962.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/wm8962.c b/sound/soc/codecs/wm8962.c
index 969479c..3ec5cc7 100644
--- a/sound/soc/codecs/wm8962.c
+++ b/sound/soc/codecs/wm8962.c
@@ -3430,6 +3430,8 @@ static int wm8962_set_fll(struct snd_soc_codec *codec, int fll_id, int source,
 			    WM8962_FLL_FRAC | WM8962_FLL_REFCLK_SRC_MASK |
 			    WM8962_FLL_ENA, fll1 | WM8962_FLL_ENA);
 
+	fll1 |= WM8962_FLL_ENA;
+
 	dev_dbg(codec->dev, "FLL configured for %dHz->%dHz\n", Fref, Fout);
 
 	ret = 0;
-- 
1.7.10.4

